app/serializers/api/variant_serializer.rb

```diff
@@ -41,10 +41,10 @@ def thumb_url
   end
 
   def unit_price_price
-    (rand * 10).round(2)
+    price_with_fees / VariantUnits::UnitPrices.new(object).unit_price_denominator
```

Comment:
IMO it'd be a nicer API as `VariantUnits::UnitPrices.new(object).denominator`. It feels a bit redundant otherwise.

Reply:
I think I must have realized the same thing at some point and changed it :)